test/metabase/server/request/util_test.clj

```diff
@@ -50,6 +52,28 @@
     nil
     nil))
 
+(deftest ip-address-test
+  (let [request (ring.mock/request :get "api/session")]
+    (testing "request with no forwarding"
+      (is (= "127.0.0.1"
+             (request.u/ip-address request))))
+    (testing "request with forwarding"
+      (let [mock-request (-> (ring.mock/request :get "api/session")
+                             (ring.mock/header "X-Forwarded-For" "5.6.7.8"))]
+        (is (= "5.6.7.8"
+               (request.u/ip-address mock-request))))
+      (testing "multiple IP addresses"
+        (let [mock-request (-> (ring.mock/request :get "api/session")
+                               (ring.mock/header "X-Forwarded-For" "1.2.3.4, 5.6.7.8"))]
```

Comment:
Not sure if it's super important, but it's probably worthwhile adding a test where two or more XFF headers exist (different than many IPs for a single XFF header), a trick occasionally used by a bad actor to confuse some network elements.

Reply:
I'm not sure we could easily write a test for that here -- by the time we see the headers they're all in a map which means we couldn't add duplicates. However duplicates are handled happens upstream somewhere -- either by Jetty (Java-land) or Ring (Clojure-land)... it's worth looking in to and seeing what happens. But if something funky is happening we'd probably have to submit an upstream patch

Reply:
I'll open a new GH issue to investigate this separately, and see if there's a way we can actually test it in Clojure-land

Reply:
https://github.com/metabase/metabase/issues/15577